Metal: fix crash by enabling concurrent compilation only if available

kemenaran opened the pull request:
Bug: https://bugs.dolphin-emu.org/issues/13404

On macOS 13.6 / Intel Iris with Metal 2, when starting any game, Dolphin crashes immediately with this message:

> -[MTLIGAccelDevice setShouldMaximizeConcurrentCompilation:]: unrecognized selector

It seems that even on macOS 13.3+, not all devices implement this selector (first implemented in #11911).

As the behavior is an optional optimization anyway, we now perform a dynamic detection to avoid crashing if the feature is not available.

kemenaran commented on Source/Core/VideoBackends/Metal/MTLMain.mm:
```diff
@@ -96,8 +96,11 @@ static bool WindowSystemTypeSupportsMetal(WindowSystemType type)
-    if (@available(macOS 13.3, *))
-      [adapter setShouldMaximizeConcurrentCompilation:YES];
+    if (@available(macOS 13.3, *)) {
+      if ([adapter respondsToSelector:@selector(setShouldMaximizeConcurrentCompilation:)]) {
```
We need the double-if there, as it seems `if(@available(…))` can only take a single condition.

TellowKrinkle replied:
Personally I'd prefer if we replaced the `@available` and disabled the associated warning with [one of these](https://github.com/dolphin-emu/dolphin/blob/937cb8ef3a0a413e9c76c0cf90cffd8a935ab595/Source/Core/VideoBackends/Metal/MTLUtil.mm#L284-L312), as the `@available` is useless if it doesn't actually work.

Also add a comment mentioning that the 13.3 check doesn't work even though the headers say it should, mentioning the exact cpu model and macOS version it fails on (get it with `sysctl machdep.cpu.brand_string`)

kemenaran replied:
Done (CPU model, comment, and the availability check).

kemenaran commented:
PR updated, with:

- Fixed linter warnings,
- Added an explanation on why this is failing when using OCLP patches, which models are affected, and a link to the OCLP doc.

TellowKrinkle reviewed (approved):
LGTM

AdmiralCurtiss commented:
Fixed the remaining lint errors.
